L1Trigger/DemonstratorTools/plugins/GTTFileReader.cc

```diff
@@ -48,61 +50,144 @@ class GTTFileReader : public edm::stream::EDProducer<> {
-// class implementation
-//
-
 GTTFileReader::GTTFileReader(const edm::ParameterSet& iConfig)
-    : fileReader_(l1t::demo::parseFileFormat(iConfig.getUntrackedParameter<std::string>("format")),
-                  iConfig.getParameter<std::vector<std::string>>("files"),
-                  kFramesPerTMUXPeriod,
-                  kVertexTMUX,
-                  kEmptyFrames,
-                  kChannelSpecs) {
-  produces<l1t::VertexWordCollection>();
+    : processOutputToCorrelator_(iConfig.getParameter<bool>("processOutputToCorrelator")),
+      processInputTracks_(iConfig.getParameter<bool>("processInputTracks")),
+      processOutputToGlobalTrigger_(iConfig.getParameter<bool>("processOutputToGlobalTrigger")),
+      kEmptyFramesOutputToCorrelator_(iConfig.getUntrackedParameter<unsigned int>("kEmptyFramesOutputToCorrelator")),
+      kEmptyFramesInputTracks_(iConfig.getUntrackedParameter<unsigned int>("kEmptyFramesInputTracks")),
+      kEmptyFramesOutputToGlobalTrigger_(
+          iConfig.getUntrackedParameter<unsigned int>("kEmptyFramesOutputToGlobalTrigger")),
+      l1VertexCollectionName_(iConfig.getParameter<std::string>("l1VertexCollectionName")),
+      l1TrackCollectionName_(iConfig.getParameter<std::string>("l1TrackCollectionName")) {
+  if (processOutputToCorrelator_) {
+    fileReaderOutputToCorrelator_ =
+        l1t::demo::BoardDataReader(l1t::demo::parseFileFormat(iConfig.getUntrackedParameter<std::string>("format")),
+                                   iConfig.getParameter<std::vector<std::string>>("filesOutputToCorrelator"),
+                                   l1t::demo::kFramesPerTMUXPeriod,
+                                   l1t::demo::kGTTBoardTMUX,
+                                   kEmptyFramesOutputToCorrelator_,
+                                   l1t::demo::kChannelMapOutputToCorrelator);
+    produces<l1t::VertexWordCollection>(l1VertexCollectionName_);
+  }
+  if (processInputTracks_) {
+    fileReaderInputTracks_ =
+        l1t::demo::BoardDataReader(l1t::demo::parseFileFormat(iConfig.getUntrackedParameter<std::string>("format")),
+                                   iConfig.getParameter<std::vector<std::string>>("filesInputTracks"),
+                                   l1t::demo::kFramesPerTMUXPeriod,
+                                   l1t::demo::kGTTBoardTMUX,
+                                   kEmptyFramesInputTracks_,
+                                   l1t::demo::kChannelMapInput);
+    produces<TTTrackCollection>(l1TrackCollectionName_);
+  }
+  if (processOutputToGlobalTrigger_) {
+    // fileReaderOutputToGlobalTrigger_ =
+    //   l1t::demo::BoardDataReader(l1t::demo::parseFileFormat(iConfig.getUntrackedParameter<std::string>("format")),
+    // 				 iConfig.getParameter<std::vector<std::string>>("filesOutputToGlobalTrigger"),
+    // 				 l1t::demo::kFramesPerTMUXPeriod,
+    // 				 l1t::demo::kGTTBoardTMUX,
+    // 				 kEmptyFramesOutputToGlobalTrigger_,
+    // 				 l1t::demo::kChannelMapInput);
```

Comment:
If this is no longer in use could you remove it?

Reply:
Rather than being unused, this is a stub for the next addition here, reading back objects from the GT buffer files. 

I do not feel extremely strongly about keeping the commented-out stub in, however; just let me know

Reply:
If it's going to be in use then it's fine to keep it. We just don't want commenting to be used where the delete button should be.